src/Microsoft.DotNet.Darc/src/Darc/Operations/AddBuildToChannelOperation.cs

```diff
@@ -178,6 +187,46 @@ private async Task<int> PromoteBuildAsync(Build build, Channel targetChannel, IR
             }
         }
 
+        private async Task<bool> ValidateAzDOBuildAsync(AzureDevOpsClient azdoClient, string azureDevOpsAccount, string azureDevOpsProject, int azureDevOpsBuildId)
+        {
+            try
+            {
+                var artifacts = await azdoClient.GetBuildArtifactsAsync(azureDevOpsAccount, azureDevOpsProject, azureDevOpsBuildId, maxRetries: 5);
+
+                // The build manifest is always necessary
+                if (!artifacts.Any(f => f.Name.Equals("AssetManifests")))
+                {
+                    Console.Write("The build that you want to promote doesn't have a Build Manifest. That's required for publishing. Aborting.");
```

Comment:
```suggestion
                    Console.Write("The build that you want to add to a new channel doesn't have a Build Manifest. That's required for publishing. Aborting.");
```

here and in the others?